plugins/nanospell/plugin.js

```diff
@@ -733,11 +736,18 @@
 				return '/spellcheck/nano/';
 			}
 
+			function scheduleRender(event) {
+				setTimeout(render.bind(this, event), 0);
+			}
+
 			function render(event) {
 				var rootElement = event.data.root,
-					needsBookmarkCreated = event.data.needsBookmarkCreated,
+					selectionStart = editor.getSelection().getStartElement(),
```

Comment:
I mentioned this before, the selection can now more easily change between scheduling of this as a callback + actually executing it.  That said, the contains call is still pretty fast (i assume browser voodoo)